### PR TITLE
Windows log fixes

### LINKED ISSUE #1228: Garbled console output on Windows
From another comment, but opening this to tackle the issue separately.

This is what I see on my Windows 7 VM

```
C:\Users\xmr\Desktop>node -v && npm -v
v6.9.2
3.10.9

C:\Users\xmr\Desktop>lighthouse --output=json --output-path=results.json https://www.bootstrapcdn.com/
  ndefined;1mLighthouse CLI Launching Chrome... ndefinedm+0ms
  ndefined;1mChromeLauncher Waiting for browser. ndefinedm+0ms
  ndefined;1mChromeLauncher Waiting for browser... ndefinedm+0ms
  ndefined;1mChromeLauncher Waiting for browser..... ndefinedm+2s
  ndefined;1mChromeLauncher Waiting for browser....... ndefinedm+2s
  ndefined;1mChromeLauncher Waiting for browser.......✓ ndefinedm+0ms
  ndefined;1mstatus Initializing… ndefinedm+2s
  ndefined;1mstatus Loading page & waiting for onload URL, HTTPS, Viewport, ThemeColor, Manifest, Accessibility, ContentWidth ndefinedm+372ms
  ndefined;1mstatusEnd Loading page & waiting for onload ndefinedm+4s
  ndefined;1mstatus Retrieving trace ndefinedm+1ms
  ndefined;1mstatus Retrieving network records ndefinedm+4s
  ndefined;1mstatus Retrieving: URL ndefinedm+0ms
  ndefined;1mstatus Retrieving: HTTPS ndefinedm+0ms
  ndefined;1mstatus Retrieving: Viewport ndefinedm+18ms
  ndefined;1mstatus Retrieving: ThemeColor ndefinedm+96ms
  ndefined;1mstatus Retrieving: Manifest ndefinedm+0ms
  ndefined;1mstatus Retrieving: Accessibility ndefinedm+746ms
  ndefined;1mstatus Retrieving: ContentWidth ndefinedm+718ms
  ndefined;1mstatus Loading page & waiting for onload ServiceWorker, Offline ndefinedm+414ms
  ndefined;1mstatusEnd Loading page & waiting for onload ndefinedm+705ms
  ndefined;1mstatus Retrieving network records ndefinedm+1ms
  ndefined;1mstatus Retrieving: ServiceWorker ndefinedm+1ms
  ndefined;1mstatus Retrieving: Offline ndefinedm+0ms
  ndefined;1mstatus Loading page & waiting for onload HTTPRedirect, HTMLWithoutJavaScript ndefinedm+351ms
  ndefined;1mstatusEnd Loading page & waiting for onload ndefinedm+3s
  ndefined;1mstatus Retrieving network records ndefinedm+0ms
  ndefined;1mstatus Retrieving: HTTPRedirect ndefinedm+0ms
  ndefined;1mstatus Retrieving: HTMLWithoutJavaScript ndefinedm+78ms
  ndefined;1mstatus Loading page & waiting for onload Styles, EventListeners, AnchorsWithNoRelNoopener, AppCacheManifest, ConsoleTimeUsage, DateNowUse, DocWriteUse, GeolocationOnStart, NotificationOnStart, TagsBlockingFirstPaint, WebSQL ndefinedm+668ms
  ndefined;1mstatusEnd Loading page & waiting for onload ndefinedm+7s
  ndefined;1mstatus Retrieving network records ndefinedm+0ms
  ndefined;1mstatus Retrieving: Styles ndefinedm+0ms
  ndefined;1mstatus Retrieving: EventListeners ndefinedm+3s
  ndefined;1mstatus Retrieving: AnchorsWithNoRelNoopener ndefinedm+888ms
  ndefined;1mstatus Retrieving: AppCacheManifest ndefinedm+0ms
  ndefined;1mstatus Retrieving: ConsoleTimeUsage ndefinedm+16ms
  ndefined;1mstatus Retrieving: DateNowUse ndefinedm+0ms
  ndefined;1mstatus Retrieving: DocWriteUse ndefinedm+62ms
  ndefined;1mstatus Retrieving: GeolocationOnStart ndefinedm+47ms
  ndefined;1mstatus Retrieving: NotificationOnStart ndefinedm+0ms
  ndefined;1mstatus Retrieving: TagsBlockingFirstPaint ndefinedm+16ms
  ndefined;1mstatus Retrieving: WebSQL ndefinedm+62ms
  ndefined;1mstatus Disconnecting from browser... ndefinedm+526ms
  ndefined;1mstatus Evaluating: Site is on HTTPS ndefinedm+56ms
  ndefined;1mstatus Evaluating: Site redirects HTTP traffic to HTTPS ndefinedm+2ms
  ndefined;1mstatus Evaluating: Has a registered Service Worker ndefinedm+1ms
  ndefined;1mstatus Evaluating: URL responds with a 200 when offline ndefinedm+1ms
  ndefined;1mstatus Evaluating: HTML has a viewport <meta> ndefinedm+3ms
  ndefined;1mstatus Evaluating: Manifest's display property is set ndefinedm+1ms
  ndefined;1mstatus Evaluating: Page contains some content when its scripts are not available ndefinedm+1ms
  ndefined;1mstatus Evaluating: First meaningful paint ndefinedm+1ms
  ndefined;1mstatus Evaluating: Perceptual Speed Index ndefinedm+30ms
  ndefined;1mstatus Evaluating: Estimated Input Latency ndefinedm+1s
  ndefined;1mstatus Evaluating: Time To Interactive (alpha) ndefinedm+962ms
  ndefined;1mstatus Evaluating: User Timing marks and measures ndefinedm+47ms
  ndefined;1mstatus Evaluating: Screenshots of all captured frames ndefinedm+412ms
  ndefined;1mstatus Evaluating: Critical Request Chains ndefinedm+283ms
  ndefined;1mstatus Evaluating: Manifest exists ndefinedm+5ms
  ndefined;1mstatus Evaluating: Manifest contains background_color ndefinedm+32ms
  ndefined;1mstatus Evaluating: Manifest contains theme_color ndefinedm+1ms
  ndefined;1mstatus Evaluating: Manifest contains icons at least 192px ndefinedm+0ms
  ndefined;1mstatus Evaluating: Manifest contains icons at least 144px ndefinedm+0ms
  ndefined;1mstatus Evaluating: Manifest contains name ndefinedm+0ms
  ndefined;1mstatus Evaluating: Manifest contains short_name ndefinedm+0ms
  ndefined;1mstatus Evaluating: Manifest's short_name won't be truncated when displayed on homescreen ndefinedm+16ms
  ndefined;1mstatus Evaluating: Manifest contains start_url ndefinedm+0ms
  ndefined;1mstatus Evaluating: HTML has a theme-color <meta> ndefinedm+0ms
  ndefined;1mstatus Evaluating: Content is sized correctly for the viewport ndefinedm+0ms
  ndefined;1mstatus Evaluating: Element aria-* roles are valid ndefinedm+0ms
  ndefined;1mstatus Evaluating: Elements with ARIA roles have the required aria-* attributes ndefinedm+0ms
  ndefined;1mstatus Evaluating: Element aria-* attributes have valid values ndefinedm+0ms
  ndefined;1mstatus Evaluating: Element aria-* attributes are valid and not misspelled or non-existent. ndefinedm+0ms
  ndefined;1mstatus Evaluating: Background and foreground colors have a sufficient contrast ratio ndefinedm+0ms
  ndefined;1mstatus Evaluating: Every image element has an alt attribute ndefinedm+0ms
  ndefined;1mstatus Evaluating: Every form element has a label ndefinedm+0ms
  ndefined;1mstatus Evaluating: No element has a tabindex attribute greater than 0 ndefinedm+0ms
  ndefined;1mstatus Evaluating: Site opens external anchors using rel="noopener" ndefinedm+0ms
  ndefined;1mstatus Evaluating: Site does not use Application Cache ndefinedm+16ms
  ndefined;1mstatus Evaluating: Page does not automatically request geolocation on page load ndefinedm+0ms
  ndefined;1mstatus Evaluating: Site does not use <link> that delay first paint ndefinedm+0ms
  ndefined;1mstatus Evaluating: Site does not use console.time() in its own scripts ndefinedm+0ms
  ndefined;1mstatus Evaluating: Site does not use Date.now() in its own scripts ndefinedm+0ms
  ndefined;1mstatus Evaluating: Site does not use document.write() ndefinedm+0ms
  ndefined;1mstatus Evaluating: Site does not use Mutation Events in its own scripts ndefinedm+0ms
  ndefined;1mstatus Evaluating: Site does not use the old CSS flexbox ndefinedm+15ms
  ndefined;1mstatus Evaluating: Site does not use WebSQL DB. ndefinedm+0ms
  ndefined;1mstatus Evaluating: Page does not automatically request notification permissions on page load ndefinedm+0ms
  ndefined;1mstatus Evaluating: Site does not use <script> in head that delays first paint ndefinedm+0ms
  ndefined;1mstatus Evaluating: Site uses HTTP/2 for its own resources ndefinedm+0ms
  ndefined;1mstatus Evaluating: Site uses passive listeners to improve scrolling performance ndefinedm+77ms
  ndefined;1mPrinter json output written to results.json ndefinedm+109ms
  ndefined;1mChromeLauncher Killing all Chrome Instances ndefinedm+0ms
```

For the `✓` and `✘` I see that `√` and `×` work. I could make a PR using those for Windows. I'm not aware of a way to detect if the font supports the characters.

### LINKED ISSUE #1261: Cannot find module 'debug/browser' when running 'npm run watch' for lighthouse-extension
I just sync the latest code, navigate to `lighthouse-extension` folder and run `npm install` (both under the root path and lighthouse-extensions path) and `npm run watch`, the following errors come, maybe a global environment issue or what else do I need to do?

```
E:\Personal\lighthouse\lighthouse-extension>npm run watch

> lighthouse@ watch E:\Personal\lighthouse\lighthouse-extension
> gulp watch

[14:51:31] Using gulpfile E:\Personal\lighthouse\lighthouse-extension\gulpfile.js
[14:51:31] Starting 'lint'...
[14:51:31] Starting 'browserify'...
[14:51:31] Starting 'browserify-lighthouse'...
[14:51:31] Starting 'html'...
[14:51:31] Finished 'html' after 764 ms
[14:51:32] Finished 'lint' after 1 s
stream.js:74
      throw er; // Unhandled stream error in pipe.
      ^

Error: Cannot find module 'debug/browser' from 'E:\Personal\lighthouse\lighthouse-core\lib'
    at E:\Personal\lighthouse\lighthouse-extension\node_modules\browser-resolve\node_modules\resolve\lib\async.js:46:17
    at process (E:\Personal\lighthouse\lighthouse-extension\node_modules\browser-resolve\node_modules\resolve\lib\async.js:173:43)
    at ondir (E:\Personal\lighthouse\lighthouse-extension\node_modules\browser-resolve\node_modules\resolve\lib\async.js:188:17)
    at load (E:\Personal\lighthouse\lighthouse-extension\node_modules\browser-resolve\node_modules\resolve\lib\async.js:69:43)
    at onex (E:\Personal\lighthouse\lighthouse-extension\node_modules\browser-resolve\node_modules\resolve\lib\async.js:92:31)
    at E:\Personal\lighthouse\lighthouse-extension\node_modules\browser-resolve\node_modules\resolve\lib\async.js:22:47
    at FSReqWrap.oncomplete (fs.js:123:15)
```

### PR DESCRIPTION
**Pull request by XhmikosR**
This is #1272 cherry picked, #1305 cherry picked and fixed.

Verified everything works on my Windows VM.

after

Closes #1272, closes #1305.
Fixes #1228, fixes #1261.

**Note to maintainers**: Do **not** squash the branch when merging in order to keep author information.

Oh, and merry Christmas 🎄 

/CC @Janpot to confirm you are OK with cherry picking your patch and close 1272.

### CODE REVIEW

**Comment by AdrianoCahete** [comment]
>  The bad news is that it appears that one or more commits were authored by someone other than the pull request submitter. We need to confirm that they're okay with their commits being contributed to this project. Please have them confirm that here in the pull request.

For me, it's ok.

**Comment by Janpot** [comment]
Not sure I agree with the proposed solution for windows. This seems to be a problem with cmd.exe  [not supporting unicode](http://stackoverflow.com/questions/388490/unicode-characters-in-windows-command-line-how) by default. I guess this change would affect anybody on windows, regardless of which shell they use. 
Nevertheless I'm in favour of pragmatic solutions so if the owners agree with this one, it's ok for me.
If you go with this one however, I think it should be abstracted away like the [figures](https://github.com/sindresorhus/figures) does. Maybe even contribute to them some of the substitutions that are missing for the tree?

**Comment by XhmikosR** [comment]
In my original patch I used figures. But being that here they don't use chalk either, I just kept the original patch @AdrianoCahete had made.

I'm not sure what you mean by abstracted. If you mean have the common code in one place, that would make sense. On the other hand, I don't see why should reinvent the wheel and not just use figures and chalk for colors.

Anyway, the fact is that this works with the default shell Windows comes with and thus fixes the issue I reported.

**Comment by XhmikosR** [comment]
@Janpot: Also, the CC here was for you to confirm you allow the use of your patch, which you didn't.

**Comment by wardpeet** [comment]
I have to agree with @Janpot that abstraction would be a better way to deal with this.
Including **figures** will add an extra dependency that we might not want/need (@paulirish)
 we could add a simple abstraction file ourselves with just icons based on runtime.

The fix works in cmd & powershell but not cygwin
image

**Comment by Janpot** [comment]
Yep, I mentioned `figures` just as an illustration of my reasoning. Whether you want an extra dependency or not is author's preference to me. In my own code I reuse as much as possible but I acknowledge there's a cost to that as well. For me the important thing is not to litter main business logic with platform switches.
I'd also keep both PRs separate as my contribution merely deals with the latest update of the `debug` package. The unicode stuff is part of a broader issue I guess.
Also want to stress this is not an ego thing or anything, if lighthouse authors want to merge this PR they totally have my ok, even if if they squash (I care about good software, not about bragging rights), just want to make sure I stated my point of view before they do so.

**Comment by wardpeet** [comment]
I would wait for @paulirish, @patrickhulce or @brendankenny to give the final judgement :)

@XhmikosR Thanks for this pr but as janpot says it are 2 seperates issues. You should fix unicode issues in this PR :)
@Janpot don't worry about it ;) it's fine.

**Comment by AdrianoCahete** [comment]
> The fix works in cmd & powershell but not cygwin

[Cygwin seems to don't show unicode by default](https://x.cygwin.com/docs/faq/cygwin-x-faq.html#q-xterm-utf8).
I never used Cygwin, I should probably be wrong.

**Comment by ebidel** [comment]
Since the other committers are ok with the cherry pick, I think we can move forward with this one. I'm going to close https://github.com/GoogleChrome/lighthouse/pull/1272 to focus efforts here.

In the future, we'd prefer if individual PRs fix individual issues. This one is getting a bit hard to follow because of the cherry picking and multiple issues that are being referenced.

File: lighthouse-cli/chrome-launcher.ts [hunk 1]
```diff
@@ -191,7 +191,11 @@ class ChromeLauncher {
         launcher
           .isDebuggerReady()
           .then(() => {
-            log.log('ChromeLauncher', waitStatus + `${green}✓${reset}`);
+            if (process.platform === 'win32') {
```
Comment (ebidel) [inline lighthouse-cli/chrome-launcher.ts]: how about:
```
// See https://github.com/GoogleChrome/lighthouse/issues/1228
const check = process.platform === 'win32' ? '\u221A' : '✓';
log.log('ChromeLauncher', waitStatus + `${green}${check}${reset}`);`
```

File: lighthouse-cli/printer.ts [hunk 2]
```diff
@@ -56,8 +56,8 @@ function formatAggregationResultItem(score: boolean | number | string, suffix?:
   const reset = '\x1B[0m';
 
   if (typeof score === 'boolean') {
-    const check = `${green}✓${reset}`;
-    const fail = `${red}✘${reset}`;
+    const check = process.platform === 'win32' ? `${green}\u221A${reset}` : `${green}✓${reset}`;
```
Comment (ebidel) [inline lighthouse-cli/printer.ts]: similar suggestion

File: lighthouse-cli/printer.ts [hunk 3]
```diff
@@ -97,7 +97,11 @@ function createOutput(results: Results, outputMode: OutputMode): string {
   let output = `\n\n${bold}Lighthouse (${version}) results:${reset} ${results.url}\n\n`;
 
   results.aggregations.forEach(aggregation => {
-    output += `▫ ${bold}${aggregation.name}${reset}\n\n`;
+    if (process.platform === 'win32') {
```
Comment (ebidel) [inline lighthouse-cli/printer.ts]: similar suggestion to dry this up.

File: lighthouse-core/formatters/critical-request-chains.js [hunk 4]
```diff
@@ -148,7 +148,11 @@ class CriticalRequestChains extends Formatter {
 
         // If the parent is the last child then don't drop the vertical bar.
         const ancestorTreeMarker = treeMarkers.reduce((markers, marker) => {
-          return markers + (marker ? '┃ ' : '  ');
+          if (process.platform === 'win32') {
+            return markers + (marker ? '\u2502 ' : '  ');
```
Comment (ebidel) [inline lighthouse-core/formatters/critical-request-chains.js]: same

File: lighthouse-cli/chrome-launcher.ts (hunk 1 above)
Comment (XhmikosR) [inline lighthouse-cli/chrome-launcher.ts]: Sounds good to me, I'll change it to that.

**Comment by XhmikosR** [comment]
@wardpeet: I don't use cygwin so I can't test there. MSYS works fine on my test VM.
@ebidel: simplified code per your comments.

Now some things to note:

1. There's code in tests that uses symbols and should also be taken care of.
2. Due to 1) it might make more sense to abstract the common parts to a separate file to include in the whole codebase.
3. If a 3rd-party lib does the job, perhaps we could use that. Oh, and see if we can make the blue lighter so that it shows better on Windows.

File: lighthouse-core/formatters/critical-request-chains.js [hunk 5]
```diff
@@ -148,7 +148,9 @@ class CriticalRequestChains extends Formatter {
 
         // If the parent is the last child then don't drop the vertical bar.
         const ancestorTreeMarker = treeMarkers.reduce((markers, marker) => {
-          return markers + (marker ? '┃ ' : '  ');
+          // See https://github.com/GoogleChrome/lighthouse/issues/1228
+          const sym = process.platform === 'win32' ? '\u2502 ' : '┃ ';
+            return markers + (marker ? sym : '  ');
```
Comment (ebidel) [inline lighthouse-core/formatters/critical-request-chains.js]: indentation is funky here.

File: lighthouse-core/formatters/critical-request-chains.js [hunk 6]
```diff
@@ -159,9 +161,14 @@ class CriticalRequestChains extends Formatter {
 
         // Create the appropriate tree marker based on the depth of this
         // node as well as whether or not it has children and is itself the last child.
-        const treeMarker = ancestorTreeMarker +
-            (isLastChild ? '┗━' : '┣━') +
+        let treeMarker = ancestorTreeMarker;
+        if (process.platform === 'win32') {
+          treeMarker += (isLastChild ? '\u2514\u2500' : '\u251C\u2500') +
+            (hasChildren ? '\u252C' : '\u2500');
```
Comment (ebidel) [inline lighthouse-core/formatters/critical-request-chains.js]: 4 space indent

File: lighthouse-core/formatters/critical-request-chains.js [hunk 7]
```diff
@@ -159,9 +161,14 @@ class CriticalRequestChains extends Formatter {
...
+        } else {
+          treeMarker += (isLastChild ? '┗━' : '┣━') +
             (hasChildren ? '┳' : '━');
```
Comment (ebidel) [inline lighthouse-core/formatters/critical-request-chains.js]: 4 space indent

**Comment by ebidel** [comment]
Can you also look into the failing tests?

File: lighthouse-core/formatters/critical-request-chains.js (hunk 6 above)
Comment (XhmikosR) [inline lighthouse-core/formatters/critical-request-chains.js]: Should it be 4 space indentation? I don't see this being enforced by ESLint ;/

**Comment by XhmikosR** [comment]
@ebidel: the tests failure is due to the expired SSL certificate

```
shop.polymer-project.org uses an invalid security certificate. The certificate expired on Σάββατο, 31 Δεκεμβρίου 2016 02:00. The current time is Σάββατο, 31 Δεκεμβρίου 2016 20:28. Error code: SEC_ERROR_EXPIRED_CERTIFICATE
```

File: lighthouse-core/formatters/critical-request-chains.js (hunk 6 above)
Comment (ebidel) [inline lighthouse-core/formatters/critical-request-chains.js]: Correct. We do it elsewhere, but sadly, I don't think there's a rule.

Comment (XhmikosR) [inline lighthouse-core/formatters/critical-request-chains.js]: This seems fine as a 2-space indentation. I added the rule locally to ESLint and it's passing.

**Comment by ebidel** [comment]
Derp, thanks for the heads up on the ssl cert. I'll tell the Polymer team.

Can you file an issue for the remaining items?

**Comment by XhmikosR** [comment]
Which issues? We mentioned many things here :p

File: lighthouse-cli/chrome-launcher.ts [hunk 8]
```diff
@@ -191,7 +191,9 @@ class ChromeLauncher {
         launcher
           .isDebuggerReady()
           .then(() => {
-            log.log('ChromeLauncher', waitStatus + `${green}✓${reset}`);
+            // See https://github.com/GoogleChrome/lighthouse/issues/1228
+            const check = process.platform === 'win32' ? '\u221A' : '✓';
```
Comment (brendankenny) [inline lighthouse-cli/chrome-launcher.ts]: can you pull this into a file-level global a la `green` and `reset`?

File: lighthouse-cli/printer.ts [hunk 9]
```diff
@@ -56,8 +56,9 @@ function formatAggregationResultItem(score: boolean | number | string, suffix?:
   const reset = '\x1B[0m';
 
   if (typeof score === 'boolean') {
-    const check = `${green}✓${reset}`;
-    const fail = `${red}✘${reset}`;
+    // See https://github.com/GoogleChrome/lighthouse/issues/1228
+    const check = process.platform === 'win32' ? `${green}\u221A${reset}` : `${green}✓${reset}`;
```
Comment (brendankenny) [inline lighthouse-cli/printer.ts]: globals here too

**Comment by ebidel** [comment]
https://github.com/GoogleChrome/lighthouse/pull/1307#issuecomment-269875523 and anything else this PR doesn't currently cover :)

File: lighthouse-core/formatters/critical-request-chains.js [hunk 10]
```diff
@@ -159,9 +161,14 @@ class CriticalRequestChains extends Formatter {
 
         // Create the appropriate tree marker based on the depth of this
         // node as well as whether or not it has children and is itself the last child.
-        const treeMarker = ancestorTreeMarker +
-            (isLastChild ? '┗━' : '┣━') +
-            (hasChildren ? '┳' : '━');
+        let treeMarker = ancestorTreeMarker;
+        if (process.platform === 'win32') {
+          treeMarker += (isLastChild ? '\u2514\u2500' : '\u251C\u2500') +
```
Comment (brendankenny) [inline lighthouse-core/formatters/critical-request-chains.js]: maybe do this as with `ancestorTreeMarker ` above, where the symbol is picked conditionally (maybe at file level) but all platforms construct the string in the same way?

**Comment by XhmikosR** [comment]
All right, I'll try to wrap everything up and make a couple of issues next year :)

File: lighthouse-core/formatters/critical-request-chains.js [hunk 11]
```diff
@@ -148,7 +148,9 @@ class CriticalRequestChains extends Formatter {
 
         // If the parent is the last child then don't drop the vertical bar.
         const ancestorTreeMarker = treeMarkers.reduce((markers, marker) => {
-          return markers + (marker ? '┃ ' : '  ');
+          // See https://github.com/GoogleChrome/lighthouse/issues/1228
+          const sym = process.platform === 'win32' ? '\u2502 ' : '┃ ';
```
Comment (brendankenny) [inline lighthouse-core/formatters/critical-request-chains.js]: file-level consts for these may work better, too.

File: lighthouse-core/lib/log.js [hunk 12]
```diff
@@ -32,13 +32,7 @@ const colors = {
...
 
 // whitelist non-red/yellow colors for debug()
-if (isBrowser) {
-  const debugBrowser = require('debug/browser');
-  debugBrowser.colors = [colors.cyan, colors.green, colors.blue, colors.magenta];
-} else {
-  const debugNode = require('debug/node');
-  debugNode.colors = [colors.cyan, colors.green, colors.blue, colors.magenta];
-}
+debug.colors = [colors.cyan, colors.green, colors.blue, colors.magenta];
```
Comment (brendankenny) [inline lighthouse-core/lib/log.js]: how does this play along with the changes @paulirish was making in #915? We don't want to regress the browserified build

Comment (brendankenny) [inline lighthouse-core/lib/log.js]: another option is to lock the `package.json` version to `debug` 2.2.0 as our `yarn.lock` file does

File: lighthouse-core/formatters/critical-request-chains.js (hunk 6 above)
Comment (brendankenny) [inline lighthouse-core/formatters/critical-request-chains.js]: > This seems fine as a 2-space indentation

definitely needs a four space indent after a continuation like this :)

Happy to have a rule to enforce that...may just be the eslint config didn't bring that over when we updated it recently.

**Comment by ebidel** [comment]
Confirming the smokehouse tests pass if you remove shop.

File: lighthouse-core/lib/log.js (hunk 12 above)
Comment (XhmikosR) [inline lighthouse-core/lib/log.js]: This PR shouldn't affect anything else other than Windows console.

File: lighthouse-core/formatters/critical-request-chains.js (hunk 10 above)
Comment (XhmikosR) [inline lighthouse-core/formatters/critical-request-chains.js]: I don't mind doing this, I"m just having trouble naming those symbols :/

File: lighthouse-cli/chrome-launcher.ts (hunk 8 above)
Comment (XhmikosR) [inline lighthouse-cli/chrome-launcher.ts]: You mean along with `green` and `reset` at the top of the file?

**Comment by XhmikosR** [comment]
Ah, I found where I saw this  https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-cli/test/smokehouse/smokehouse.js#L36

So, ideally, after this PR goes in, all common relevant stuff should go into the same file to prevent WET.

File: lighthouse-core/lib/log.js (hunk 12 above)
Comment (paulirish) [inline lighthouse-core/lib/log.js]: yeah this simplification shoulda been in 915. lgtm

File: lighthouse-cli/chrome-launcher.ts [hunk 13]
```diff
@@ -30,6 +30,12 @@ const log = require('../lighthouse-core/lib/log');
 const spawn = childProcess.spawn;
 const execSync = childProcess.execSync;
 
+const green = '\x1B[32m';
+const reset = '\x1B[0m';
+
+// See https://github.com/GoogleChrome/lighthouse/issues/1228
+const tick = process.platform === 'win32' ? '\u221A' : '✓';
```
Comment (paulirish) [inline lighthouse-cli/chrome-launcher.ts]: Can we simplify and just always use the unicode escape, and never the literal?

```js
const tick = '\u221A'; // ✓
```

(I'm assuming the windows character in play is the exact same as the literal... that right?)

Comment (brendankenny) [inline lighthouse-cli/chrome-launcher.ts]: looks like they're different. ✓ vs √

Comment (paulirish) [inline lighthouse-cli/chrome-launcher.ts]: okay. then we'll stick with this hilarity. 

Let's pull `process.platform === 'win32'` out into a `isWindows` boolean then
 @XhmikosR , sg?

**Comment by Janpot** [comment]
@paulirish just want to point out it's not just a simplification. `debug` changed their directory structure and `require('debug/browser')` simply doesn't resolve to any file anymore.

File: lighthouse-cli/chrome-launcher.ts (hunk 13 above)
Comment (XhmikosR) [inline lighthouse-cli/chrome-launcher.ts]: Yeah, unfortunately, the normal tick is bold and looks a lot better on normal terminals, i.e. not Windows :/

I'll add the `const` now and this can be refactored later, see https://github.com/GoogleChrome/lighthouse/pull/1307#issuecomment-269895427

**Comment by paulirish** [comment]
Seriously great group effort on this. Thanks all!
